### PR TITLE
Document interpolation variables on dpl v2 pages

### PR DESCRIPTION
**Pull request by svenfuchs**
Addresses https://github.com/travis-ci/docs-travis-ci-com/issues/2675

### CODE REVIEW

**Comment by adrianbroher** [comment]
Hi @svenfuchs, thanks for the prompt response.  That change helps tremendously.  However I would like to suggest two improvements:

```
The following variable are availabe for interpolation on `commit_message`:
```

Does this mean Ruby string interpolation?   The [dpl code states](https://github.com/travis-ci/dpl/blob/91e405cdf2b45ee692787d81d09cb887c2d2ae19/lib/dpl/helper/interpolate.rb#L7-L9)

```
# Variables can be contained in scripts, shell commands, and messages.
# They have the syntax `%{name}` or `%s` (or any other identifier supported
# by [Kernel#sprintf](https://ruby-doc.org/core-2.6.3/Kernel.html#method-i-format)).
```

So the placeholders are sprintf format specifiers and not string interpolation?  The code itself is over my head as I don't write Ruby so I can't answer that myself.

There is commit https://github.com/travis-ci/dpl/commit/ea6bd6633056abb3cfe31ea7504babfd7374b27d that allows the interpolation of environment variables.  It would be nice to document the correct syntax to access those.  From what I understand the syntax is:

```
# assuming `export FOO=bar`
commit_message: "Doing %{$FOO}"
# Would result in "Doing bar"
```

**Comment by svenfuchs** [comment]
@adrianbroher yes, hmm, and no.

The inline documentation you link to is meant for internal use mostly. The syntax `%s` would not be available to users.

The syntax is `%{variable-name}` though, yes.

While `"Doing %{$FOO}"` should work for users as well, one can simply use Bash interpolation instead, i.e. `"Doing ${FOO}"` or just `"Doing $FOO"`. These will be replaced when the `dpl` Bash command is executed that Travis CI will embed to your build script when the `deploy` section is present in your build configuration.

You are right that this should be elaborated more in the documentation. I have added https://github.com/travis-ci/docs-travis-ci-com/pull/2677/commits/2e144b493e599b959fba5a13e3b3330ba63a096d which I think should address this.

Thanks again!

**Comment by adrianbroher** [comment]
@svenfuchs Thanks for clarifying the syntax and updating the docs.

File: _includes/deploy/providers/codedeploy.md [hunk 1]
```diff
@@ -52,4 +52,38 @@ For example, `access_key_id` can be given as
 
 * `AWS_ACCESS_KEY_ID=<access_key_id>` or 
 * `CODEDEPLOY_ACCESS_KEY_ID=<access_key_id>`
+## Interpolation variables
+
+The following variable are availabe for interpolation on `description`:
```
Comment (Pezi777) [inline _includes/deploy/providers/codedeploy.md]:
```suggestion
The following variables are available for interpolation on `description`:
```

File: _includes/deploy/providers/flynn.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+
+
+{{ include.content }}
+{% endunless %}
+
+## Status
+
+Support for deployments to Flynn is in **development**. Please see [Maturity Levels](/user/deployment-v2#maturity-levels) for details.
+## Known options
```
Comment (Pezi777) [inline _includes/deploy/providers/flynn.md]:
```suggestion

## Known options
```

File: _includes/deploy/providers/lambda.md [hunk 3]
```diff
@@ -59,4 +59,38 @@ For example, `access_key_id` can be given as
 
 * `AWS_ACCESS_KEY_ID=<access_key_id>` or 
 * `LAMBDA_ACCESS_KEY_ID=<access_key_id>`
+## Interpolation variables
+
+The following variable are availabe for interpolation on `description`:
```
Comment (Pezi777) [inline _includes/deploy/providers/lambda.md]:
```suggestion
The following variables are available for interpolation on `description`:
```

File: lib/dpl/docs.rb [hunk 4]
```diff
@@ -303,6 +308,44 @@ def env # TODO
       end
     end
 
+    class Vars < Struct.new(:cmd)
+      STR = <<~str
+        ## Interpolation variables
+
+        The following variable are availabe for interpolation on %s:
```
Comment (Pezi777) [inline lib/dpl/docs.rb]:
```suggestion
        The following variables are available for interpolation on %s:
```

Comment (svenfuchs) [reply lib/dpl/docs.rb]: fixed. thanks 👍

**Pull request merged**